File: app/src/main/java/com/jobtick/activities/NotificationSettings.java
package com.jobtick.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;

import androidx.annotation.Nullable;

import com.google.android.material.appbar.MaterialToolbar;
import com.jobtick.R;
import android.annotation.SuppressLint;

import com.jobtick.widget.ExtendedSettingItem;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class NotificationSettings extends ActivityBase {


    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.toolbar)
    MaterialToolbar toolbar;

    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.email_notifications)
    ExtendedSettingItem rtlBtnEmailNotification;

    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.push_notifications)
    ExtendedSettingItem rtlBtnPushNotifications;

    @SuppressLint("NonConstantResourceId")
    @BindView(R.id.sms_notifications)
    ExtendedSettingItem rtlBtnSMSNotifications;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_notification_settings);
        ButterKnife.bind(this);
        initToolbar();
    }

    private void initToolbar() {
        toolbar.setNavigationIcon(R.drawable.ic_back);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setTitle("Notification Settings");
    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                break;
        }

        return super.onOptionsItemSelected(item);
    }


    @OnClick({R.id.email_notifications, R.id.push_notifications, R.id.sms_notifications})
    public void onViewClicked(View view) {
        Intent intent;
        switch (view.getId()) {
            case R.id.email_notifications:
                intent = new Intent(NotificationSettings.this, EmailNotificationsSettingsActivity.class);
                startActivity(intent);
                break;
            case R.id.push_notifications:
                intent = new Intent(NotificationSettings.this, PushNotificationsSettingsActivity.class);
                startActivity(intent);
                break;
            case R.id.sms_notifications:
                intent = new Intent(NotificationSettings.this, SMSNotificationsSettingsActivity.class);
                startActivity(intent);
                break;
        }
    }
}

File: app/src/main/java/com/jobtick/android/activities/NotificationSettings.kt
package com.jobtick.android.activities

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import com.google.android.material.appbar.MaterialToolbar
import com.jobtick.android.R
import com.jobtick.android.widget.ExtendedSettingItem

class NotificationSettings : ActivityBase() {
    private lateinit var toolbar: MaterialToolbar
    private lateinit var rtlBtnEmailNotification: ExtendedSettingItem
    private lateinit var rtlBtnPushNotifications: ExtendedSettingItem
    private lateinit var rtlBtnSMSNotifications: ExtendedSettingItem

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_notification_settings)
        initIDS()
        initToolbar()
        onViewClick()
    }

    private fun initIDS() {
        toolbar = findViewById(R.id.toolbar)
        rtlBtnEmailNotification = findViewById(R.id.email_notifications)
        rtlBtnEmailNotification.setDrawableID(R.drawable.ic_mail_v2)
        rtlBtnPushNotifications = findViewById(R.id.push_notifications)
        rtlBtnPushNotifications.setDrawableID(R.drawable.ic_notif_setting)
        rtlBtnSMSNotifications = findViewById(R.id.sms_notifications)
        rtlBtnSMSNotifications.setDrawableID(R.drawable.ic_sms)
    }

    private fun initToolbar() {
        toolbar.setNavigationIcon(R.drawable.ic_back)
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.title = "Notification Settings"
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> onBackPressed()
        }
        return super.onOptionsItemSelected(item)
    }

    private fun onViewClick() {
        var intent: Intent
        rtlBtnEmailNotification.setOnClickListener {
            intent = Intent(this@NotificationSettings, EmailNotificationsSettingsActivity::class.java)
            startActivity(intent)
        }
        rtlBtnPushNotifications.setOnClickListener {
            intent = Intent(this@NotificationSettings, PushNotificationsSettingsActivity::class.java)
            startActivity(intent)
        }
        rtlBtnSMSNotifications.setOnClickListener {
            intent = Intent(this@NotificationSettings, SMSNotificationsSettingsActivity::class.java)
            startActivity(intent)
        }
    }
}